pkg/shell.go

```diff
@@ -83,34 +90,76 @@ func ExecShell(req TerminalSize, path string) {
 	}
 }
 
-func createWebsocketConn(req interface{}, path string) (*websocket.Conn, error) {
-	command, err := query.Values(req)
-	if err != nil {
-		return nil, err
-	}
+func createWebsocketConn(ctx context.Context, req interface{}, path string) (*websocket.Conn, error) {
+	var wsConn *websocket.Conn
 
-	wsURL, err := url.Parse(fmt.Sprintf("%s%s", utils.WebsocketUrl(), path))
-	if err != nil {
-		return nil, err
-	}
-	pattern := regexp.MustCompile("%5B([0-9]+)%5D=")
-	wsURL.RawQuery = pattern.ReplaceAllString(command.Encode(), "[${1}]=")
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf("context deadline exceeded: %w", ctx.Err())
+		default:
+		}
+		command, err := query.Values(req)
+		if err != nil {
+			return nil, err
+		}
 
-	tokenType, token, err := utils.GetAccessToken()
-	if err != nil {
-		return nil, err
-	}
+		wsURL, err := url.Parse(fmt.Sprintf("%s%s", utils.WebsocketUrl(), path))
+		if err != nil {
+			return nil, err
+		}
 
-	headers := http.Header{"Authorization": {utils.GetAuthorizationHeaderValue(tokenType, token)}}
-	wsConn, _, err := websocket.DefaultDialer.Dial(wsURL.String(), headers)
-	if err != nil {
-		return nil, err
+		pattern := regexp.MustCompile("%5B([0-9]+)%5D=")
+		wsURL.RawQuery = pattern.ReplaceAllString(command.Encode(), "[${1}]=")
+
+		tokenType, token, err := utils.GetAccessToken()
+		if err != nil {
+			return nil, err
+		}
+
+		headers := http.Header{"Authorization": {utils.GetAuthorizationHeaderValue(tokenType, token)}}
+		wsConn, _, err = websocket.DefaultDialer.Dial(wsURL.String(), headers)
+		if err != nil {
+			log.Printf("WebSocket dial failed: %v. Retrying in 5s...\n", err)
+			time.Sleep(5 * time.Second)
+			continue
+		}
+
+		// Setup ping/pong handlers
+		wsConn.SetReadLimit(512)
```

Comment:
From the doc, it will configure the client to force close the cnx is a websocket msg is greather than 512 bytes.

https://pkg.go.dev/github.com/gorilla/websocket#Conn.SetReadLimit


Which is nok ;x